src/test/java/lambda/part3/exercise/Mapping.java

```diff
@@ -407,12 +407,11 @@ public boolean forNext(Consumer<R> c) {
                 @Override
                 public boolean forNext(Consumer<R> c) {
                     boolean result = currentIterable.forNext(c);
+                    boolean isNotEnd = true;
 
-                    if (!result) {
-                        boolean isNotEnd = self.forNext(getNewIterable);
-                        if (isNotEnd) {
-                            result = currentIterable.forNext(c);
-                        }
+                    while (!result && isNotEnd) {

```

Comment:
on [line 525](https://github.com/java8-course/lambda/pull/120/files/9bdebd0f9eff56aab3c93cd2ed8e0975f1bf2b4c#diff-813ac2502b8b33c07ee7111af3dd7543R525) where is such a test. And I also have tested different combinations of empty strings with flatMap. Can't get a clue what is wrong with this method.